pkg/util/util.go

```diff
@@ -215,22 +215,52 @@ func ComponentResourceRemoval(ctx context.Context, c client.Client, obj interfac
 
 	logger.Info("Removing resource", "name", resource.GetName(), "namespace", resource.GetNamespace(), "GVK", resource.GetObjectKind().GroupVersionKind(), "dryRun", dryRun)
 
-	err = c.Get(ctx, types.NamespacedName{Name: resource.GetName(), Namespace: resource.GetNamespace()}, resource)
+	ok, err := getResourceForDeletion(ctx, c, resource, logger)
+	if !ok {
+		return err
+	}
+
+	if !shouldDeleteResource(resource, hcoName, logger) {
+		return nil
+	}
+
+	opts := getDeletionOption(dryRun, wait)
+
+	if deleted, err := doDeleteResource(ctx, c, resource, opts); !deleted {
```

Comment:
I would implement doDeleteResource such that it only returns error. If it is not nil, just return it. Otherwise, continue.